2022-09-18 20:31:51,557 - kedro.framework.session.session - INFO - Kedro project continuous_machine_learning
2022-09-18 20:31:51,594 - py.warnings - WARNING - /Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.16.5 and <1.23.0 is required for this version of SciPy (detected version 1.23.1
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"

2022-09-18 20:31:55,465 - kedro.io.data_catalog - INFO - Loading data from 'companies' (CSVDataSet)...
2022-09-18 20:31:55,498 - kedro.pipeline.node - INFO - Running node: preprocess_companies_node: preprocess_companies([companies]) -> [data_processing.preprocessed_companies]
2022-09-18 20:31:55,523 - kedro.io.data_catalog - INFO - Saving data to 'data_processing.preprocessed_companies' (ParquetDataSet)...
2022-09-18 20:31:55,547 - kedro.runner.sequential_runner - INFO - Completed 1 out of 9 tasks
2022-09-18 20:31:55,547 - kedro.io.data_catalog - INFO - Loading data from 'shuttles' (ExcelDataSet)...
2022-09-18 20:32:01,049 - kedro.pipeline.node - INFO - Running node: preprocess_shuttles_node: preprocess_shuttles([shuttles]) -> [data_processing.preprocessed_shuttles]
2022-09-18 20:32:01,063 - py.warnings - WARNING - /Users/ryojikn/Downloads/GitHub/continuous_machine_learning/src/spaceflights/pipelines/data_processing/nodes.py:15: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  x = x.str.replace("$", "").str.replace(",", "")

2022-09-18 20:32:01,092 - kedro.io.data_catalog - INFO - Saving data to 'data_processing.preprocessed_shuttles' (ParquetDataSet)...
2022-09-18 20:32:01,133 - kedro.runner.sequential_runner - INFO - Completed 2 out of 9 tasks
2022-09-18 20:32:01,134 - kedro.io.data_catalog - INFO - Loading data from 'data_processing.preprocessed_shuttles' (ParquetDataSet)...
2022-09-18 20:32:01,288 - kedro.io.data_catalog - INFO - Loading data from 'data_processing.preprocessed_companies' (ParquetDataSet)...
2022-09-18 20:32:01,295 - kedro.io.data_catalog - INFO - Loading data from 'reviews' (CSVDataSet)...
2022-09-18 20:32:01,328 - kedro.pipeline.node - INFO - Running node: create_model_input_table_node: create_model_input_table([data_processing.preprocessed_shuttles,data_processing.preprocessed_companies,reviews]) -> [model_input_table]
2022-09-18 20:32:02,772 - kedro.io.data_catalog - INFO - Saving data to 'model_input_table' (ParquetDataSet)...
2022-09-18 20:32:03,163 - kedro.runner.sequential_runner - INFO - Completed 3 out of 9 tasks
2022-09-18 20:32:03,164 - kedro.io.data_catalog - INFO - Loading data from 'model_input_table' (ParquetDataSet)...
2022-09-18 20:32:03,293 - kedro.io.data_catalog - INFO - Loading data from 'params:data_science.active_modelling_pipeline.model_options' (MemoryDataSet)...
2022-09-18 20:32:03,294 - kedro.pipeline.node - INFO - Running node: split_data_node: split_data([model_input_table,params:data_science.active_modelling_pipeline.model_options]) -> [data_science.active_modelling_pipeline.X_train,data_science.active_modelling_pipeline.X_test,data_science.active_modelling_pipeline.y_train,data_science.active_modelling_pipeline.y_test]
2022-09-18 20:32:03,356 - kedro.io.data_catalog - INFO - Saving data to 'data_science.active_modelling_pipeline.X_train' (MemoryDataSet)...
2022-09-18 20:32:03,360 - kedro.io.data_catalog - INFO - Saving data to 'data_science.active_modelling_pipeline.X_test' (MemoryDataSet)...
2022-09-18 20:32:03,362 - kedro.io.data_catalog - INFO - Saving data to 'data_science.active_modelling_pipeline.y_train' (MemoryDataSet)...
2022-09-18 20:32:03,364 - kedro.io.data_catalog - INFO - Saving data to 'data_science.active_modelling_pipeline.y_test' (MemoryDataSet)...
2022-09-18 20:32:03,377 - kedro.runner.sequential_runner - INFO - Completed 4 out of 9 tasks
2022-09-18 20:32:03,378 - kedro.io.data_catalog - INFO - Loading data from 'model_input_table' (ParquetDataSet)...
2022-09-18 20:32:03,483 - kedro.io.data_catalog - INFO - Loading data from 'params:data_science.candidate_modelling_pipeline.model_options' (MemoryDataSet)...
2022-09-18 20:32:03,484 - kedro.pipeline.node - INFO - Running node: split_data_node: split_data([model_input_table,params:data_science.candidate_modelling_pipeline.model_options]) -> [data_science.candidate_modelling_pipeline.X_train,data_science.candidate_modelling_pipeline.X_test,data_science.candidate_modelling_pipeline.y_train,data_science.candidate_modelling_pipeline.y_test]
2022-09-18 20:32:03,528 - kedro.io.data_catalog - INFO - Saving data to 'data_science.candidate_modelling_pipeline.X_train' (MemoryDataSet)...
2022-09-18 20:32:03,531 - kedro.io.data_catalog - INFO - Saving data to 'data_science.candidate_modelling_pipeline.X_test' (MemoryDataSet)...
2022-09-18 20:32:03,533 - kedro.io.data_catalog - INFO - Saving data to 'data_science.candidate_modelling_pipeline.y_train' (MemoryDataSet)...
2022-09-18 20:32:03,534 - kedro.io.data_catalog - INFO - Saving data to 'data_science.candidate_modelling_pipeline.y_test' (MemoryDataSet)...
2022-09-18 20:32:03,548 - kedro.runner.sequential_runner - INFO - Completed 5 out of 9 tasks
2022-09-18 20:32:03,548 - kedro.io.data_catalog - INFO - Loading data from 'data_science.active_modelling_pipeline.X_train' (MemoryDataSet)...
2022-09-18 20:32:03,551 - kedro.io.data_catalog - INFO - Loading data from 'data_science.active_modelling_pipeline.y_train' (MemoryDataSet)...
2022-09-18 20:32:03,553 - kedro.pipeline.node - INFO - Running node: train_model_node: train_model([data_science.active_modelling_pipeline.X_train,data_science.active_modelling_pipeline.y_train]) -> [data_science.active_modelling_pipeline.regressor]
2022-09-18 20:32:03,767 - kedro.io.data_catalog - INFO - Saving data to 'data_science.active_modelling_pipeline.regressor' (PickleDataSet)...
2022-09-18 20:32:03,771 - kedro.runner.sequential_runner - INFO - Completed 6 out of 9 tasks
2022-09-18 20:32:03,771 - kedro.io.data_catalog - INFO - Loading data from 'data_science.candidate_modelling_pipeline.X_train' (MemoryDataSet)...
2022-09-18 20:32:03,774 - kedro.io.data_catalog - INFO - Loading data from 'data_science.candidate_modelling_pipeline.y_train' (MemoryDataSet)...
2022-09-18 20:32:03,775 - kedro.pipeline.node - INFO - Running node: train_model_node: train_model([data_science.candidate_modelling_pipeline.X_train,data_science.candidate_modelling_pipeline.y_train]) -> [data_science.candidate_modelling_pipeline.regressor]
2022-09-18 20:32:03,799 - kedro.io.data_catalog - INFO - Saving data to 'data_science.candidate_modelling_pipeline.regressor' (PickleDataSet)...
2022-09-18 20:32:03,802 - kedro.runner.sequential_runner - INFO - Completed 7 out of 9 tasks
2022-09-18 20:32:03,802 - kedro.io.data_catalog - INFO - Loading data from 'data_science.active_modelling_pipeline.regressor' (PickleDataSet)...
2022-09-18 20:32:03,803 - kedro.io.data_catalog - INFO - Loading data from 'data_science.active_modelling_pipeline.X_test' (MemoryDataSet)...
2022-09-18 20:32:03,805 - kedro.io.data_catalog - INFO - Loading data from 'data_science.active_modelling_pipeline.y_test' (MemoryDataSet)...
2022-09-18 20:32:03,806 - kedro.pipeline.node - INFO - Running node: evaluate_model_node: evaluate_model([data_science.active_modelling_pipeline.regressor,data_science.active_modelling_pipeline.X_test,data_science.active_modelling_pipeline.y_test]) -> None
2022-09-18 20:32:03,851 - spaceflights.pipelines.data_science.nodes - INFO - Model has a coefficient R^2 of 0.462 on test data.
2022-09-18 20:32:03,854 - kedro.runner.sequential_runner - INFO - Completed 8 out of 9 tasks
2022-09-18 20:32:03,854 - kedro.io.data_catalog - INFO - Loading data from 'data_science.candidate_modelling_pipeline.regressor' (PickleDataSet)...
2022-09-18 20:32:03,855 - kedro.io.data_catalog - INFO - Loading data from 'data_science.candidate_modelling_pipeline.X_test' (MemoryDataSet)...
2022-09-18 20:32:03,860 - kedro.io.data_catalog - INFO - Loading data from 'data_science.candidate_modelling_pipeline.y_test' (MemoryDataSet)...
2022-09-18 20:32:03,861 - kedro.pipeline.node - INFO - Running node: evaluate_model_node: evaluate_model([data_science.candidate_modelling_pipeline.regressor,data_science.candidate_modelling_pipeline.X_test,data_science.candidate_modelling_pipeline.y_test]) -> None
2022-09-18 20:32:03,874 - spaceflights.pipelines.data_science.nodes - INFO - Model has a coefficient R^2 of 0.449 on test data.
2022-09-18 20:32:03,888 - kedro.runner.sequential_runner - INFO - Completed 9 out of 9 tasks
2022-09-18 20:32:03,891 - kedro.runner.sequential_runner - INFO - Pipeline execution completed successfully.
2022-09-18 20:32:08,392 - py.warnings - WARNING - /Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.16.5 and <1.23.0 is required for this version of SciPy (detected version 1.23.1
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"

2022-09-18 20:32:13,142 - uvicorn.error - ERROR - Exception in ASGI application
Traceback (most recent call last):
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/uvicorn/protocols/http/httptools_impl.py", line 372, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/uvicorn/middleware/message_logger.py", line 82, in __call__
    raise exc from None
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/uvicorn/middleware/message_logger.py", line 78, in __call__
    await self.app(scope, inner_receive, inner_send)
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/fastapi/applications.py", line 199, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc from None
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc from None
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/starlette/routing.py", line 580, in __call__
    await route.handle(scope, receive, send)
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/starlette/routing.py", line 241, in handle
    await self.app(scope, receive, send)
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/starlette/routing.py", line 52, in app
    response = await func(request)
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/fastapi/routing.py", line 227, in app
    response_data = await serialize_response(
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/fastapi/routing.py", line 113, in serialize_response
    response_content = _prepare_response_content(
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/site-packages/fastapi/routing.py", line 95, in _prepare_response_content
    return dataclasses.asdict(res)
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/dataclasses.py", line 1075, in asdict
    return _asdict_inner(obj, dict_factory)
  File "/Users/ryojikn/opt/miniconda3/envs/tf_macos/lib/python3.9/dataclasses.py", line 1082, in _asdict_inner
    value = _asdict_inner(getattr(obj, f.name), dict_factory)
AttributeError: 'DataNodeMetadata' object has no attribute 'plot'
